src/lib/viewers/image/ImageViewer.js

```diff
@@ -165,6 +187,56 @@ class ImageViewer extends ImageBaseViewer {
         };
     }
 
+    /**
+     * Noop for handling annotation creation event.
+     *
+     * @override
+     * @return {void}
+     */
+    handleAnnotationCreateEvent() {}
+
+    /**
+     * Noop for handling annotation creator change event.
+     *
+     * @override
+     * @return {void}
+     */
+    handleAnnotationCreatorChangeEvent() {}
+
+    /**
+     * Gets the viewport dimensions.
+     *
+     * @return {Object} the width & height of the viewport
+     */
+    getViewportDimensions() {
+        return {
+            width: this.wrapperEl.clientWidth - 2 * IMAGE_PADDING,
+            height: this.wrapperEl.clientHeight - 2 * IMAGE_PADDING,
+        };
+    }
+
+    /**
+     * Sets mode to be AnnotationMode.NONE if the zoomed image overflows the viewport.
+     *
+     * @return {void}
+     */
+    handleZoomEvent({ newScale, type }) {
+        const [width, height] = newScale;
+        const isUserInitiated = type !== undefined;
```

Comment:
Why the existence of `type` is connected with `isUserInitiated`? So if we accidentally change `type` parameter in `zoom` event in the future, this logic might be broken then?

Reply:
Right now there are two possibilities of `handleZoomEvent` being called.

1. On initial render, `handleZoomEvent` is called (even though the user has not zoomed).
2. The user performs a zoom action (type may be `in` or `out`.

If we change the type parameter to some other string, since that is still a user action, this logic would not break because we are checking against whether or not a user has performed an action.

Reply:
How about
```
// explanation
if (!type) {
    return;
}
```

Reply:
Seems like this comment is still outstanding